webapp/src/pages/RegisterPage/RegisterPage.tsx

```diff
@@ -0,0 +1,161 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import {
+  Button,
+  ButtonVariant,
+  Input,
+  InputVariant,
+  Text,
+  TextColor,
+  TextVariant,
+  TextWeight,
+} from '../../shared/components';
+import { ResponseError } from '../../shared/generated';
+import { authApi } from '../../shared/services/ApiService';
+import { SubmitStatus } from '../../shared/types';
+import { LOGIN_OPTIONS_URL } from '../../shared/urls';
+import './RegisterPage.css';
+
+export default function RegisterPage() {
+  const [username, setUsername] = useState('');
```

Comment:
couldn't we use something like this:
```
const [userRegistrationDto, setUserRegistrationDto] = useState<RegisterUserDto>({username: '', fullName: '', email: '', password: '', confirmationPassword: ''});
```
instead of these some 5 useState, since we already have the type anyway?

Reply:
I've finally used `useReducer` instead. [https://reactjs.org/docs/hooks-reference.html#usereducer](https://reactjs.org/docs/hooks-reference.html#usereducer)

Reply:
To be honest I think this is not the _by the book_ ideal use case for useReducer (a reducer becomes a real help, and almost a necessity, when we have to handle complex state logic. Here we don't have complex state logic: we're only updating the dto).

My idea was something like what you can find here `webapp/src/shared/hooks/UseDrag.tsx` L21.

But it works, so I'll just approve it.